src/libcore/iter.rs

```diff
@@ -280,6 +289,16 @@ pub trait Iterator {
     /// different sized integer, the `zip` function provides similar
     /// functionality.
     ///
+    /// # Undefined overflow
+    ///
+    /// The method does no guarding against overflows, so enumerating more than
+    /// `usize::MAX` elements is undefined.
```

Comment:
I second @alexcrichton   's note here.  No reason to be putting these sorts of scary terms here, IMHO